lua/mason-null-ls/settings.lua

```diff
@@ -0,0 +1,32 @@
+local M = {}
+
+---@class MasonNullLsSettings
+local DEFAULT_SETTINGS = {
+	-- A list of sources to automatically install if they're not already installed. Example: { "stylua" }
+	-- This setting has no relation with the `automatic_installation` setting.
+	ensure_installed = {},
+
+	-- NOTE: this is left here for future porting in case needed
+	-- Whether sources that are set up (via null-ls) should be automatically installed if they're not already installed.
+	-- This setting has no relation with the `ensure_installed` setting.
+	-- Can either be:
+	--   - false: Servers are not automatically installed.
+	--   - true: All servers set up via lspconfig are automatically installed.
+	--   - { exclude: string[] }: All servers set up via lspconfig, except the ones provided in the list, are automatically installed.
+	--       Example: automatic_installation = { exclude = { "rust_analyzer", "solargraph" } }
+	-- automatic_installation = false,
```

Comment:
I am not in favor of disabling this feature as it is working and it seems there was at least 1 user who was using this plugin solely because of it. any reason we can't keep it as the logic is already there for it

Reply:
I'm not in favor of disabling it as well, I've mentioned that I'll look into re-implementing it back ;3

Reply:
Are you okay with getting this changes merged into a feature branch? I can work on adding back this missing feature and then I will merge into `main`

Reply:
not at all, that would be great!